tensilelite/Tensile/KernelHelperNaming.py

```diff
@@ -0,0 +1,262 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+################################################################################
+
+from copy import deepcopy
+from enum import IntEnum
+
+from Tensile.Common.GlobalParameters import internalParameters
+from Tensile.KernelWriterBetaOnly import KernelWriterBetaOnly
+from Tensile.KernelWriterConversion import KernelWriterConversion
+from Tensile.KernelWriterActivationEnumHeader import KernelWriterActivationEnumHeader
+from Tensile.KernelWriterActivationFunction import KernelWriterActivationFunction
+from Tensile.KernelWriterActivationOnly import KernelWriterActivationOnly
+from Tensile.KernelWriterReduction import KernelWriterReduction
+
+
+class KernelHelperEnum(IntEnum):
+    BetaOnly = 0
+    Conversion = 1
+    ActivationEnumHeader = 2
+    ActivationFunction = 3
+    ActivationOnly = 4
+    Reduction = 5
+
+
+def conversionKernelObjectsNames(solution):
+  # need to check that the fields mutated in the the init function aren't used in the naming function
+  # e.g. usebias got set to zero if gradient which changes behavior of name function.
+  conversionKernelObjectsNames = []
+  load_vector_width = [1, 2] if solution["ProblemType"]["DataType"].isDouble() else [1, 2, 4]
+  gsuList = [internalParameters["GlobalSplitUPGR"]]
+  if solution["GlobalSplitUAlgorithm"] == "SingleBuffer":
+    gsuList = [1]
+  elif solution["GlobalSplitUAlgorithm"] == "MultipleBufferSingleKernel":
+    return
+  for vw in load_vector_width:
+    # for _ in gsuList: I don't think this does anything
+    if solution["ProblemType"]["UseBias"]:
+      typeList = solution["ProblemType"]["BiasDataTypeList"]
+      if solution["ProblemType"]["Gradient"]:
+      #  # If gradient + bias D, generates a normal GSU kernel for bias D = nullptr case
+        conversionKernelObjectsNames.append(KernelWriterConversion._getKernelName(solution, vw))
+      for btype in typeList:
+        conversionKernelObjectsNames.append(KernelWriterConversion._getKernelName(solution, vw, btype))
+    else:
+      conversionKernelObjectsNames.append(KernelWriterConversion._getKernelName(solution, vw))
+  return conversionKernelObjectsNames if conversionKernelObjectsNames else []
+
+
+def activationEnumHeaderObjectsNames(solution):
+  activationEnumHeaderObjectsNames = []
+  if solution["ProblemType"]["ActivationType"] in ['all', 'hipblaslt_all']:
+    activationEnumHeaderObjectsNames.append(KernelWriterActivationEnumHeader._getKernelName(solution))
+  return activationEnumHeaderObjectsNames
```

Comment:
This could be a one-liner
```suggestion
def activationEnumHeaderObjectsNames(solution):
  return [] if solution["ProblemType"]["ActivationType"] not in ['all', 'hipblaslt_all'] else [KernelWriterActivationEnumHeader._getKernelName(solution)]
```
edit: I see similar, but slightly more complex functions below that may not be oneline-able. If they can't all be made oneliners, then I support consistency, as it is coded now.